repository.py

```diff
@@ -0,0 +1,543 @@
+#!/usr/bin/env python
+
+# Taken from https://opendev.org/openstack/sunbeam-charms/src/commit/5b37e0a6919668f23b8c7b148717714889fd4381/repository.py
+
+import logging
+import argparse
+import dataclasses
+import pathlib
+import shutil
+import subprocess
+import os
+import glob
+
+import yaml
+
+ROOT_DIR = pathlib.Path(__file__).parent
+EXTERNAL_LIB_DIR = ROOT_DIR / "external" / "lib"
+BUILD_PATH = ROOT_DIR / ".slurm-build"
```

Comment:
See my earlier comment about using `_build` instead. Curious to know your thoughts on this.